pkg/roachpb/BUILD.bazel

```diff
@@ -223,15 +158,53 @@ go_proto_library(
     importpath = "github.com/cockroachdb/cockroach/pkg/roachpb",
     proto = ":roachpb_proto",
     visibility = ["//visibility:public"],
+    go_srcs = [
+        "api.go",
```

Comment:
Are ALL of these sources necessary to get the target to compile? I'd prefer the list be kept as small as possible (unless they're really all interdependent).

Reply:
Hm. I'm not immediately sure about that. If anything, I'd argue that we should consider switching _every go package_ that contains proto-grenerated go sources to pass _any_ additional sources, whether or not the proto-generated sources depend on them, this way.

In Go, the logical unit of dependency, of compilation, etc is the package. Having one Go package that consists of multiple bazel targets is not the norm, and makes for a confusing and brittle development workflow IMO. You could easily imagine that you have `a.go` in the `go_library` and then `b.go` in the `go_proto_library`, because `foo.proto` uses a `FooID` casttype defined in b.go, but then you add another message that has a casttype for something defined in a.go... it is not immediately clear that you've done something wrong, since a.go and b.go are side by side, the sum target actually still builds just fine, but you'd violated some rule that the pb target builds on its own. 

I think conceptually it'd be simpler to use if we just kept 1 go package = 1 bazel target, so if that package has go and proto-generated go sources -- even if they _could_ be split into a DAG of independent targets -- then it is still always  just one target, a go_proto_library that passes all its go raw sources in go_srcs?